packages/atomic/src/components/search/atomic-pager/atomic-pager.spec.ts

```diff
@@ -0,0 +1,497 @@
+  });
+
+  it('should disable the previous button when there is no previous page', async () => {
+    await renderPager({
+      pagerState: {hasPreviousPage: false},
+    });
+
+    await expect.element(locators.previous).toHaveAttribute('disabled');
+  });
+
+  it('should disable the next button when there is no next page', async () => {
+    await renderPager({
+      pagerState: {hasNextPage: false},
+    });
+
+    await expect.element(locators.next).toHaveAttribute('disabled');
+  });
+
+  describe('when clicking on the previous button', () => {
+    let focusSpy: MockInstance;
+    let eventSpy: MockInstance;
+    let previousSpy: MockInstance;
+    let element: AtomicPager;
+
+    beforeEach(async () => {
+      element = await renderPager({
+        pagerState: {hasPreviousPage: true, currentPage: 2},
+      });
+      focusSpy = vi.spyOn(
+        element.bindings.store.state.resultList!,
+        'focusOnFirstResultAfterNextSearch'
+      );
+      eventSpy = vi.spyOn(element, 'dispatchEvent');
+      previousSpy = vi.spyOn(element.pager, 'previousPage');
+
+      await locators.previous.click();
+    });
+
+    it('should call #focusOnFirstResultAfterNextSearch', async () => {
+      expect(focusSpy).toHaveBeenCalled();
+    });
+
+    it("should dispatch 'atomic/scrollToTop'", async () => {
+      expect(eventSpy).toHaveBeenCalledWith(
+        new CustomEvent('atomic/scrollToTop')
+      );
+    });
+
+    it('should call #pager.previousPage', async () => {
+      expect(previousSpy).toHaveBeenCalled();
+    });
+
+    it('should not be disabled when there is a previous page', async () => {
+      await expect.element(locators.previous).not.toHaveAttribute('disabled');
+    });
+  });
+
+  describe('when clicking on the next button', () => {
+    let focusSpy: MockInstance;
+    let eventSpy: MockInstance;
+    let nextSpy: MockInstance;
+    let element: AtomicPager;
+
+    beforeEach(async () => {
+      element = await renderPager({
+        pagerState: {hasNextPage: true, currentPage: 1},
+      });
+      focusSpy = vi.spyOn(
+        element.bindings.store.state.resultList!,
+        'focusOnFirstResultAfterNextSearch'
+      );
+      eventSpy = vi.spyOn(element, 'dispatchEvent');
+      nextSpy = vi.spyOn(element.pager, 'nextPage');
+
+      await locators.next.click();
+    });
+
+    it('should call #focusOnFirstResultAfterNextSearch', async () => {
+      expect(focusSpy).toHaveBeenCalled();
+    });
+
+    it("should dispatch 'atomic/scrollToTop'", async () => {
+      expect(eventSpy).toHaveBeenCalledWith(
+        new CustomEvent('atomic/scrollToTop')
+      );
+    });
+
+    it('should call #pager.nextPage', async () => {
+      expect(nextSpy).toHaveBeenCalled();
+    });
+
+    it('should not be disabled when there is a next page', async () => {
+      await expect.element(locators.next).not.toHaveAttribute('disabled');
+    });
+  });
+
+  describe('when clicking on a page button', () => {
```

Comment:
Similar comments as for the "when clicking on the previous button" describe block.